### DISCUSSION TITLE
Error when trying to download CS Tools to my laptop

### DISCUSSION THREAD

**Discussion by DavinRK1** [thoughtspot/cs_tools#139] [General] [answered]
Hello,

I am following the directions on https://thoughtspot.github.io/cs_tools/tutorial/install/#__tabbed_2_1 this webpage to attempt to download cs tools to my windows laptop using Windows PowerShell. I copy and pasted the following command into the PowerShell Window: powershell -c "IRM https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py | python - --install"

My output was as follows:
```
IRM : Unable to connect to the remote server
At line:1 char:1
+ IRM https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_ ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

Is a binary install what is needed in this situation? If that is the case here is the output from python -m sysconfig:
```

Platform: "win-amd64"
Python version: "3.12"
Current installation scheme: "nt"
Paths:
        data = "C:\Program Files\Python312"
        include = "C:\Program Files\Python312\Include"
        platinclude = "C:\Program Files\Python312\Include"
        platlib = "C:\Program Files\Python312\Lib\site-packages"
        platstdlib = "C:\Program Files\Python312\Lib"
        purelib = "C:\Program Files\Python312\Lib\site-packages"
        scripts = "C:\Program Files\Python312\Scripts"
        stdlib = "C:\Program Files\Python312\Lib"
```

**Comment by boonhapus** [comment]
Hey @DavinRK1 it looks like your company may require a proxy to download files outside the network, I can provide an offline binary for you in a couple of days!

**Reply by DavinRK1** [reply]
Great thank you @boonhapus!

Quick question are these offline binary installs unique? Or can multiple people on my team use the same install I am using?

Thanks,
Davin

**Reply by boonhapus** [reply]
@DavinRK1 they're built based on the combination of your system architecture, specifically these two elements. It's meant to provide you with the machine-optimized dependencies of what allows cs tools to run.

```
Platform: "win-amd64"
Python version: "3.12"
```

Also, do you plan to use any particular **[Syncer](https://thoughtspot.github.io/cs_tools/syncer/what-is/)**? This is how CS Tools can write data back to a database or specialized file format (other than CSV). I'll need to include these in your offline binary as well.

**Reply by DavinRK1** [reply]
@boonhapus Thanks for that reply makes sense. I will plan to use syncer so please do include that in the offline binary. Thank you!

**Reply by boonhapus** [reply]
@DavinRK1 which syncer do you plan to use? Snowflake, BigQuery, etc? Please let me know, thanks!

**Reply by DavinRK1** [reply]
Snowflake! Thank you.

**Reply by DavinRK1** [reply]
Hi @boonhapus I just wanted to message and make sure you aren't waiting on any information from me, please let me know if you are. Thanks!

**Reply by boonhapus** [reply, marked as answer]
Hey @DavinRK1 -- apologies this is so late! Can you try the steps below?

- Extract the zip file.
-  Run python against the `_bootstrapper.py` file with `--offline-mode` specified

`cd cs-tools_1.5.8b7_win_amd64_3.12`
`python _bootstrapper.py --install --no-clean --offline-mode .`

It's a bit too big for GitHub discussions, so I've attached an Egnyte link. Let me know if that works for you!
https://thoughtspot.egnyte.com/dl/cYNQXlYV2N

**Reply by DavinRK1** [reply]
This worked its looking like I have CS tools downloaded on the laptop now, thank you!
